website/docs/API.md

```diff
@@ -515,22 +517,94 @@ Defined as:
 ```jsx
 function waitFor<T>(
   expectation: () => T,
-  { timeout: number = 4500, interval: number = 50 }
+  { timeout: number = 1000, interval: number = 50 }
 ): Promise<T> {}
 ```
 
-Waits for non-deterministic periods of time until your element appears or times out. `waitFor` periodically calls `expectation` every `interval` milliseconds to determine whether the element appeared or not.
+Should be used when testing asynchronous code or code with timers. `waitFor` allows to wait for some expectation (e.g. the presence of an element on the screen). The `expectation` param is a callback that should throw until what you're waiting for has realized.
+
+For instance, this is how we'd wait for an element to appear:
+
+```tsx
```

Comment:
I think we should avoid cases for `waitFor` that uses queries, as users should be rather using `findBy*` queries rather than wrapping `getBy*` queries in `waitFor`. Ref; https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

Reply:
Agreed, I have updated the examples